src/ion/data_structures.rs

```diff
@@ -544,3 +599,17 @@ pub struct Stats {
     pub halfmoves_count: usize,
     pub edits_count: usize,
 }
+
+// Helper function for generating sorting keys. The order of arguments is from
+// the most significant field to the least significant one.
+//
+// These work best when the fields are stored in reverse order in memory so that
+// they can be loaded with a single u64 load.
```

Comment:
s/loaded with a single u64 load/loaded with a single u64 load on little-endian machines/

(makes it a bit clearer why this optimization works as well)